src/DataTypes/Serializations/SerializationUUID.cpp

```diff
@@ -44,11 +45,44 @@ void SerializationUUID::serializeTextQuoted(const IColumn & column, size_t row_n
 
 void SerializationUUID::deserializeTextQuoted(IColumn & column, ReadBuffer & istr, const FormatSettings &) const
 {
-    UUID x;
-    assertChar('\'', istr);
-    readText(x, istr);
-    assertChar('\'', istr);
-    assert_cast<ColumnUUID &>(column).getData().push_back(x);    /// It's important to do this at the end - for exception safety.
+    UUID uuid;
+    bool fast = false;
+    if (istr.available() >= 38)
+    {
+        assertChar('\'', istr);
+        char * next_pos = find_first_symbols<'\\', '\''>(istr.position(), istr.buffer().end());
+        size_t len = next_pos - istr.position();
+        if ((len == 32) && (istr.position()[32] == '\''))
+        {
+            parseUUIDWithoutSeparator(
+                reinterpret_cast<const UInt8 *>(istr.position()), std::reverse_iterator<UInt8 *>(reinterpret_cast<UInt8 *>(&uuid) + 16));
+            istr.ignore(33);
+            fast = true;
+        }
+        else if ((len == 36) && (istr.position()[36] == '\''))
+        {
+            parseUUID(
+                reinterpret_cast<const UInt8 *>(istr.position()), std::reverse_iterator<UInt8 *>(reinterpret_cast<UInt8 *>(&uuid) + 16));
+            istr.ignore(37);
+            fast = true;
+        }
+        else
+        {
+            // It's ok to go back in the position because we haven't read from the buffer except the first char
+            // and we know there were at least 38 bytes available (so no new read has been triggered)
+            istr.position()--;
+        }
+    }
+
+    if (!fast)
+    {
+        String quoted_chars;
+        readQuotedStringInto<false>(quoted_chars, istr);
+        ReadBufferFromString parsed_quoted_buffer(quoted_chars);
+        readText(uuid, parsed_quoted_buffer);
+    }
+
+    assert_cast<ColumnUUID &>(column).getData().push_back(std::move(uuid)); /// It's important to do this at the end - for exception safety.
```

Comment:
maybe something like this?
```suggestion
    assertChar('\'', istr);

    UUID uuid;
    char * begin = reinterpret_cast<char *>(&uuid);
    char * pos = begin;
    bool high = true;
    bool delim = false;
    for (size_t n = 0; !istr.eof() && 16 > pos - begin; ++istr.position(), ++n)
    {
        if (*istr.position() == '\'')
            break;
        if (n == 8 && *istr.position() == '-')
        {
            delim = true;
            continue;
        }
        if (delim && (n == 12 || n == 16 || n == 20))
        {
            if (*istr.position() != '-')
                break;
            continue;
        }

        char c = unhex(*istr.position());
        if (c > 0x0f)
            throw ParsingException("Cannot parse UUID", ErrorCodes::CANNOT_PARSE_UUID);

        if (high)
            *pos = c << 4;
        else
        {
            *pos &= c;
            ++pos;
        }
        high = !high;
    }

    if (pos - begin != 16)
        throw ParsingException("Cannot parse UUID", ErrorCodes::CANNOT_PARSE_UUID);

    assertChar('\'', istr);

    assert_cast<ColumnUUID &>(column).getData().push_back(uuid);    /// It's important to do this at the end - for exception safety.
```

Reply:
2 things:
- Performance of that is really bad. It goes from 3.3 seconds with the PR to 16.605 seconds, so it'd need to be really improved.
- I think it'd need to replace all `readText` / `readCSV` / `readUUIDTextImpl` and other read UUID functions in ReadHelpers, not just this implementation.

Tested with CH local:
```
Select * from generateRandom('u UUID') limit 100000000 INTO OUTFILE '/tmp/uuid_100m.values';
```

And then:
```
Select * from generateRandom('u UUID') limit 100000000 INTO OUTFILE '/tmp/uuid_100m.values';
```

Reply:
Would it make sense to merge this or something equivalent and then work on other PR to improve hex correctness (on all possible input formats of UUID)?

Reply:
wow, 16sec... interesting...
ok, I think it makes sense to merge your variant first

Reply:
Updated the comment above because I had pasted the sql to generate the file twice, instead of adding the benchmark query.

One idea about perf: it might be worth having this check only once in `parseUUID` and `parseUUIDWithoutSeparator` (or even in `parseHex`). Verifying that the whole string only includes some characters (0-9a-fA-F) should be much faster if done at once with a known size (as the compiler can and will likely introduce simd operations) rather than doing it char by char.

Reply:
yes, I think it's reasonable solution
Frankly it always was speed against reliability trade off... actually I have such byte-by-byte reader in IPv6 parser - now I'm thinking about improving it also...